### ISSUE TITLE
`color()` does not apply tint to models

### ISSUE THREAD

**Issue by SK83RJOSH** [KiwiOrg/Kiwi#114]
I'm trying to keep the golf game mode simple, and assign pseudo-random colors to each player using `color()` components on each of their player objects. It works on text, which is awesome, but the description of color implies it should work on anything.

Right now I get the following result (the balls and arrows should be tinted):
![image](https://user-images.githubusercontent.com/1371787/219951370-5b0ea6ad-ae3c-40db-9809-738bf51697d5.png)

**Comment by SK83RJOSH** [comment]
[ball.zip](https://github.com/KiwiOrg/Kiwi/files/10777022/ball.zip)

**Comment by SK83RJOSH** [comment]
It seems that `color()` is not applied to objects spawned via AssetUrl/Asynchronously. Setting the component every frame fixes this issue, but, isn't particularly viable in the long run. Investigating a bit further to see if I can figure out where this will need to be handle.

**Comment by FredrikNoren** [comment]
@SK83RJOSH I'm guessing that it's spawning the ball as a hierarchy of entities then, and the color is only on the root element? Can you dump the client side ecs hierarchy?

**Comment by SK83RJOSH** [comment]
@FredrikNoren Nope, not at the moment, I've kept the ball/arrow/line separate so I can rotate and scale them all independently from one another; so this is reproducible simply by spawning an object with either `from_url()` component with a `color()` attached.

I dumped the server hierarchy yesterday, and they all had mesh() + color() set; but the color wasn't applying. My working theory is the `gpu_ecs` misses the `changed` event and uses a default somewhere.

**Comment by SK83RJOSH** [comment]
Aha, okay I ended up spotting the issue, the clients never receive the latest color:
![image](https://user-images.githubusercontent.com/1371787/220070933-9db4f67d-3e17-43a1-a15f-127c68af356b.png)


[hierarchy.zip](https://github.com/KiwiOrg/Kiwi/files/10783039/hierarchy.zip)

**Comment by SK83RJOSH** [comment]
I've noticed when running via `kiwi run` the loading models are rendered as blue (which is not the correct color) and after loading the color gets set to white. After a bit more triaging, I've found the culprit:
https://github.com/AmbientOrg/Ambient/blob/main/crates/model/src/model.rs#L456

Not exactly sure what the correct behavior here should be, the root should have `color()` set; and I can't explain the desync between client and server.
